devices/apa102/apa102_test.go

```diff
@@ -698,21 +693,25 @@ func BenchmarkWriteColorful(b *testing.B) {
 	_, _ = d.Write(pixels[:])
 	b.ResetTimer()
 	for i := 0; i < b.N; i++ {
+		d.Intensity = uint8(i)
+		d.Temperature = uint16((3000 + i) & 0x1FFF)
 		_, _ = d.Write(pixels[:])
 	}
 }
 
-func BenchmarkDrawNRGBAColorful(b *testing.B) {
-	// Takes the fast path.
-	b.ReportAllocs()
-	img := image.NewNRGBA(image.Rect(0, 0, 150, 1))
-	for i := range img.Pix {
-		img.Pix[i] = uint8(i) + uint8(i>>8)
+func benchmarkDraw(b *testing.B, o Opts, img image.Image, f fn) {
+	switch im := img.(type) {
```

Comment:
A generic way to do:
```
for x := 0; x < bounds.Dx(); x++ {
  pix := fn(i)
  c : = color.NRGBA{R: pix[0], G: pix[1], B:pix[2], A:255}
  img.Set(0, x, c)
}
```

since setup time is not a big deal here.

Reply:
Done.  Though I'm also allowing for a non-zero y even though the current tests don't use any.